packages/arb-util/fireblocks/fireblocks.go

```diff
@@ -0,0 +1,512 @@
+	Pending              string `json:"pending"`
+	LockedAmount         string `json:"lockedAmount"`
+	TotalStakedCPU       string `json:"totalStakedCPU"`
+	SelfStakedCPU        string `json:"selfStakedCPU"`
+	SelfStakedNetwork    string `json:"selfStakedNetwork"`
+	PendingRefundCPU     string `json:"pendingRefundCPU"`
+	PendingRefundNetwork string `json:"pendingRefundNetwork"`
+}
+
+type TransactionRequestDestination struct {
+	Amount      string `json:"amount"`
+	Destination string `json:"destination"`
+}
+
+type fireblocksClaims struct {
+	Uri      string `json:"uri"`
+	Nonce    int64  `json:"nonce"`
+	Iat      int64  `json:"iat"`
+	Exp      int64  `json:"exp"`
+	Sub      string `json:"sub"`
+	BodyHash string `json:"bodyHash"`
+	jwt.StandardClaims
+}
+
+func New(assetId string, baseUrl string, sourceType accounttype.AccountType, sourceId string, apiKey string, signKey *rsa.PrivateKey) *Fireblocks {
+	rand.Seed(time.Now().UnixNano())
+	return &Fireblocks{
+		apiKey:     apiKey,
+		assetId:    assetId,
+		baseUrl:    baseUrl,
+		signKey:    signKey,
+		sourceId:   sourceId,
+		sourceType: sourceType,
+	}
+}
+
+func (fb *Fireblocks) ListTransactions() (*[]TransactionDetails, error) {
+	resp, err := fb.getRequest("/v1/transactions", url.Values{})
+	if err != nil {
+		return nil, err
+	}
+
+	var result []TransactionDetails
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &result, err
+}
+
+func (fb *Fireblocks) ListVaultAccounts() (*[]VaultAccount, error) {
+	resp, err := fb.getRequest("/v1/vault/accounts", url.Values{})
+	if err != nil {
+		return nil, err
+	}
+
+	var result []VaultAccount
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return nil, err
+	}
+
+	return &result, err
+}
+
+func (fb *Fireblocks) CreateNewContractCall(destinationType accounttype.AccountType, destinationId string, destinationTag string, callData string) (*CreateTransactionResponse, error) {
+	return fb.CreateNewTransaction(destinationType, destinationId, destinationTag, "0", operationtype.ContractCall, callData)
+}
+
+func (fb *Fireblocks) CreateNewTransaction(destinationType accounttype.AccountType, destinationId string, destinationTag string, amount string, operation operationtype.OperationType, callData string) (*CreateTransactionResponse, error) {
+
+	body := &CreateNewTransactionBody{
+		AssetId:         fb.assetId,
+		Source:          TransferPeerPath{Type: fb.sourceType, Id: fb.sourceId},
+		Destination:     *NewDestinationTransferPeerPath(destinationType, destinationId, destinationTag),
+		Amount:          amount,
+		Operation:       operation,
+		ExtraParameters: TransactionExtraParameters{ContractCallData: callData},
+	}
+
+	resp, err := fb.postRequest("/v1/transactions", url.Values{}, body)
+	if err != nil {
+		return nil, err
+	}
+
+	var result CreateTransactionResponse
+	response, err := ioutil.ReadAll(resp.Body)
+	fmt.Println("Response: ", string(response))
```

Comment:
I'm guessing this was some leftover debugging code. This should probably be moved to a proper logger.

Reply:
:+1: